qpid-jms-client/src/test/java/org/apache/qpid/jms/transports/netty/NettyTcpTransportTest.java

```diff
@@ -394,6 +403,88 @@ public void testZeroSizedSentNoErrors() throws Exception {
         assertTrue(data.isEmpty());
     }
 
+    @Test(timeout = 60 * 1000, expected = IllegalStateException.class)
+    public void testCannotDeferenceSharedClosedEventLoopGroup() throws Exception {
+        try (NettyEchoServer server = createEchoServer(createServerOptions())) {
+
+            server.start();
+
+            int port = server.getServerPort();
+            URI serverLocation = new URI("tcp://localhost:" + port);
+            final TransportOptions sharedTransportOptions = createClientOptions();
+            sharedTransportOptions.setUseKQueue(false);
+            sharedTransportOptions.setUseEpoll(false);
+            sharedTransportOptions.setSharedEventLoopThreads(1);
+            Transport nioTransport = createConnectedTransport(serverLocation, sharedTransportOptions);
+            EventLoopGroupRef groupRef = getGroupRef(nioTransport);
+            assertNotNull(groupRef.ref());
+
+            nioTransport.close();
+
+            server.stop();
+
+            groupRef.ref();
+            fail();
+        }
+
+        assertTrue(!transportClosed);  // Normal shutdown does not trigger the event.
+        assertTrue(exceptions.isEmpty());
+        assertTrue(data.isEmpty());
+    }
+
+    @Test(timeout = 60 * 1000)
+    public void testSharedEventLoopGroups() throws Exception {
+        try (NettyEchoServer server = createEchoServer(createServerOptions())) {
+            server.start();
+
+            int port = server.getServerPort();
+            URI serverLocation = new URI("tcp://localhost:" + port);
+            final TransportOptions sharedTransportOptions = createClientOptions();
+            sharedTransportOptions.setUseKQueue(false);
+            sharedTransportOptions.setUseEpoll(false);
+            sharedTransportOptions.setSharedEventLoopThreads(1);
+            Transport nioTransport = createConnectedTransport(serverLocation, sharedTransportOptions);
+            Transport sharedNioTransport = createConnectedTransport(serverLocation, sharedTransportOptions);
```

Comment:
Very confusingly named, suggests one is shared and one isnt, when actually both are, as this and later bits of the test show.